pallets/order-book/src/lib.rs

```diff
@@ -420,6 +466,76 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		#[pallet::call_index(4)]
+		#[pallet::weight(T::Weights::create_order())]
+		pub fn add_trading_pair(
+			origin: OriginFor<T>,
+			asset_in: T::AssetCurrencyId,
+			asset_out: T::AssetCurrencyId,
+			min_order: T::Balance,
+		) -> DispatchResult {
+			T::AdminOrigin::ensure_origin(origin)?;
+
+			// We do not check, we just overwrite as this is an admin action.
+			TradingPairInOut::<T>::insert(asset_in.clone(), asset_out.clone(), min_order);
+			TradingPairOutIn::<T>::insert(asset_out.clone(), asset_in.clone(), ());
+
+			Self::deposit_event(Event::<T>::TradingPairAdded {
+				asset_in,
+				asset_out,
+				min_order,
+			});
+
+			Ok(())
+		}
+
+		#[pallet::call_index(5)]
+		#[pallet::weight(T::Weights::create_order())]
+		pub fn rm_trading_pair(
+			origin: OriginFor<T>,
+			asset_in: T::AssetCurrencyId,
+			asset_out: T::AssetCurrencyId,
+		) -> DispatchResult {
+			T::AdminOrigin::ensure_origin(origin)?;
+
+			// We do not check, we just remove as this is an admin action.
+			TradingPairInOut::<T>::remove(asset_in.clone(), asset_out.clone());
+			TradingPairOutIn::<T>::remove(asset_out.clone(), asset_in.clone());
```

Comment:
If we remove a trading pair, we should also cancel all order for these pairs IMO. Removing the trading pair disables placing new orders but still allows old ones to be fulfilled.

Reply:
Aren't we also fine with just letting them live till fulfillment or cancel? This would make benching this really hard

Reply:
> Aren't we also fine with just letting them live till fulfillment or cancel?

IMO, for now that's totally fine but in the future we need to weight the user experience as well. I just realized that trading pairs are not necessarily undirected edges of a graph, e.g., killing one trading pair `A -> B` does not imply counter orders `B -> A` can still be placed. 

Otherwise (neither `A -> B` nor `B -> A` can be placed), we can assume none of these orders can be fulfilled and the user has to cancel them. This made me realize that we also need a hook for cancelled orders, other wise Foreign Investments are stuck or corresponding `InvestState`/`RedeemState`s even buggy.

Reply:
Ah, you are right... 

Then I remove the other mapping. But I for now we can assume that there is rather a `placeOrder(a, b)` and a `fulfillOrder(a, b)` flow. So for swapping from `A -> B`, only `(A,B)` needs to be a valid pair.

Reply:
How should the hook look liike? I can add it to this one here.